app/models/notification.rb

```diff
@@ -15,7 +15,7 @@ def self.for(recipient, opts={})
   end
 
   def self.notify(recipient, target, actor)
-    return nil unless target.respond_to? :notification_type
+    return nil unless target.respond_to? :notification_type || recipient != actor
```

Comment:
Just reshare any public post and comment it. You should see a notification saying that you commented your own reshare. Reproduced on diaspora-fr.org. diaspote is not up to date.